nailgun/entities.py

```diff
@@ -1295,6 +1295,22 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
             entity, attrs, ignore, params)
 
 
+class GCEComputeResource(AbstractComputeResource):
+    """A representation of a Google Compute Resource entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'email': entity_fields.StringField(required=True),
+            'key_path': entity_fields.StringField(required=True),
+            'project': entity_fields.StringField(required=True),
```

Comment:
apidoc shows email, key_path, project as optional... required=true needed for them?

Reply:
Yes, All of them are absolutely mandatory.

Reply:
Mandatory for GCE, but not for others. So probably that's the reason APIDoc says it's optional.